### DISCUSSION TITLE
Why is Pypy better than CPython for socketify if it is a C++ library under the hood ?

### DISCUSSION THREAD

**Discussion by Napolitain** [cirospaciari/socketify.py#196] [Q&A] [answered]
benchmarks shows pypy faster, while it should speed up only pure python code right?

**Comment by cirospaciari** [comment, marked as answer]
CFFI at the time was the fastest integration with PyPy and compatible with CPython but it has a lot of overhead, native extensions can be implemented to favor one or another, That's why I'm reworking this without CFFI using HPy with has less overhead and will allow to CPython to be way faster just by cutting the CFFI overhead.

Also we implemented a lot of Python code, asyncio is also faster in PyPy and PyPy can interop faster with native than CPython
